plutus-tx/src/PlutusTx/Builtins/Internal.hs

```diff
@@ -372,7 +374,7 @@ that you want to be representable on-chain.
 For off-chain usage, there are conversion functions 'builtinDataToData' and
 'dataToBuiltinData', but note that these will not work on-chain.
 -}
-data BuiltinData = BuiltinData PLC.Data
+data BuiltinData = BuiltinData PLC.Data deriving Data
```

Comment:
I'm not sure how I feel about these. They're supposed to be very opaque. But I guess this is fine, since it's only for off-chain usage...

Reply:
What is the trick that makes the instances of `Show`, `Eq`, `Ord` ok? The same argument should make the `Data` instance ok?

Reply:
C.f. the comment in the issue w.r.t. this.